backend/src/users/users.controller.ts

```diff
@@ -68,4 +86,50 @@ export class UsersController {
   remove(@Param('id') id: string): Promise<DeleteResult> {
     return this.usersService.remove(+id);
   }
+
+  @Post(':id/avatar')
+  @UseInterceptors(FileInterceptor('file'))
+  @ApiBearerAuth()
+  @ApiResponse({ status: 201, description: 'Success' })
+  @ApiConsumes('multipart/form-data')
+  @ApiBody({
+    description: 'Avatar Photo',
+    type: AvatarUploadDto,
+  })
+  uploadAvatar(
+    @Param('id') id: string,
+    @UploadedFile() file: Express.Multer.File,
+  ) {
+    return this.usersService.updateAvatar(
+      +id,
+      `${this.configService.get<string>('UPLOADS_PATH')}` + file.filename,
+    );
+  }
+
+  @Get(':id/avatar')
+  @ApiBearerAuth()
+  @ApiResponse({ status: 200, description: 'Success' })
+  @ApiResponse({ status: 204, description: 'No avatar found' })
+  @ApiResponse({ status: 400, description: 'User does not exist' })
+  async getAvatar(
+    @Param('id') id: string,
+    @Response({ passthrough: true }) res,
+  ): Promise<StreamableFile> {
+    try {
+      const file = await this.usersService.getAvatar(+id);
+      res.set({
+        'Content-Type': `image/${file.ext}`,
+        'Content-Disposition': `inline; filename="avatar.${file.ext}"`,
```

Comment:
```suggestion
        'Content-Disposition': 'inline',
```
I think that in the case of inline files, the value `filename` is unused. See [this Mozilla page](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition).